python/ray/util/sgd/v2/backends/backend.py

```diff
@@ -55,20 +59,140 @@ def start(self, initialization_hook: Optional[Callable[[], None]] = None):
             self.worker_group.execute(initialization_hook)
         self._backend.on_start(self.worker_group, self._backend_config)
 
-    def run(self, train_func: Callable[[], T]) -> List[T]:
-        """Executes a training function on all workers.
+    def start_training(self, train_func: Callable[[], T]) -> None:
+        """Executes a training function on all workers in a separate thread.
```

Comment:
Can we reference the other functions that should be called after this (e.g. `fetch_next_result` and `finish_training`?

Reply:
Only `finish_training` needs to be called after this. I added this as a comment.